src/nvim/extmark.c

```diff
@@ -156,12 +159,16 @@ void extmark_set(buf_T *buf, uint32_t ns_id, uint32_t *idp, int row, colnr_T col
       buf->b_virt_line_blocks++;
     }
     if (decor_has_sign(decor)) {
+      if (revised) {
+        decor->sign_mod_id = sign_add_id;
+      }
+      decor->sign_add_id = sign_add_id++;
```

Comment:
How much stuff would break if this was reduced to a single id (which would be incremented regardless if the sign is newly added or modified)?

Reply:
Arguably not that much. When signs are first placed in the signlist currently they are sorted by `lnum` -> `priority` -> `sign_id`. Then when modified signs take precedence over other signs with the same `sign_id`.
https://github.com/neovim/neovim/blob/5e5f5174e3faa862a9bc353aa7da41487911140b/src/nvim/sign.c#L309-L312

Tested e.g. here:
https://github.com/neovim/neovim/blob/5e5f5174e3faa862a9bc353aa7da41487911140b/test/old/testdir/test_signs.vim#L1512-L1515

Reply:
IMO we should change this and just keep track of the last modified time. This is the kind of bug-for-bug compatibility which we don't aim at supporting.

Reply:
To be clear, we then want signs to be sorted by `lnum`->`priority`->`sign_id`->`last_modifed`?

Reply:
>  then want signs to be sorted by lnum->priority->sign_id->last_modifed?

Sounds reasonable